### PR TITLE
test: update editor-level tests

### PR DESCRIPTION
**Pull request by connorhaugh**
To complete https://openedx.atlassian.net/browse/TNL-9601

### CODE REVIEW

File: src/editors/Editor.test.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+const props = {
+  courseId: 'course-v1:edX+DemoX+Demo_Course',
+  blockType: 'html',
+  blockId: 'block-v1:edX+DemoX+Demo_Course+type@html+block@030e35c4756a4ddc8d40b95fbbfff4d4',
+  studioEndpointUrl: 'fakeurl.com',
+  initialize: jest.fn(),
+};
+
+jest.mock('./data/redux', () => ({
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:17]: you should not be mocking these out, but rather mocking out the hooks that trigger them.  This module has no need to mock any redux behavior.

File: src/editors/Editor.test.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      isFailed: jest.fn((state, params) => ({ isFailed: { state, params } })),
+      isFinished: jest.fn((state, params) => ({ isFailed: { state, params } })),
+    },
+  },
+}));
+
+describe('Editor', () => {
+  describe('snapshots', () => {
+    test('renders no editor when ref isnt ready', () => {
+      prepareEditorRefSpy.mockImplementationOnce(
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:42]: with the above changes, this becomes:
```
hooks.initializeApp.mockReturnValueOnce({
  editorRef: null,
  refReady: false,
  setEditorRef: jest.fn().mockName('hooks.initializeApp.setEditorRef'),
})
```
I gave setEditorRef a mockName, as it is rendered in the snapshot.

File: src/editors/Editor.test.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { shallow } from 'enzyme';
+import { Editor, mapDispatchToProps } from './Editor';
+import { thunkActions } from './data/redux';
+import * as hooks from './hooks';
+
+const prepareEditorRefSpy = jest.spyOn(hooks, 'prepareEditorRef');
+
+const props = {
+  courseId: 'course-v1:edX+DemoX+Demo_Course',
+  blockType: 'html',
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:11]: should import the blockTypes from constants to avoid "magic" strings

File: src/editors/hooks.test.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  });
+  describe('prepareEditorRef', () => {
+    let output;
+    beforeEach(() => {
+      output = module.prepareEditorRef();
+    });
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+    test('sets refReady to false by default, ref is null', () => {
```
Comment (muselesscreator) [inline src/editors/hooks.test.jsx:45]: should have a test that specifically calls out the full set of outputs
"returns an object containing editorRef, refReady, and setEditorRef values"

below that, you can then have the tests of 
* "refReady defaults to false, editorRef defaults to null"
* "when useEffect triggers, refReady is set to true"
* "calling setEditorRef updates the editorRef value"

File: src/editors/hooks.test.jsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      expect(spy).toHaveBeenCalledWith(destination);
+    });
+  });
+  describe('saveBlock', () => {
+    const mockSaveFn = jest.fn(val => val);
+    const navSpy = jest.spyOn(module, 'navigateCallback');
+    const getContentMock = jest.fn();
+    const mockEditorRef = { current: { getContent: getContentMock } };
+    const url = 'rEtUrNUrl';
+    test('it calls the save function with correct arguements', () => {
```
Comment (muselesscreator) [inline src/editors/hooks.test.jsx:88]:
```
const mockNavCallback = (returnUrl) => ({ navigateCallback: returnUrl });
const navSpy = jest.spyOn(module, 'navigateCallback').mockImplementationOnce(mockNavCallback);
const content = { some: 'content' };
const args = {
  editorRef: { current: { getContent: () => content } },
  returnUrl: 'rEtUrNUrl',
  saveFunction: jest.fn(),
});  
module.saveBlock(args);
expect(args.saveFunction).toHaveBeenCalledWith({
  returnToUnit: mockNavCallback(args.returnUrl),
  content,
});
```

**Comment by connorhaugh** [comment]
@muselesscreator ready for another round of review.

**Comment by connorhaugh** [comment]
Ready for another round of review at your earliest convenience @muselesscreator

File: src/editors/Editor.test.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  describe('snapshots', () => {
+    test('renders no editor when ref isnt ready', () => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: null, refReady: false, setEditorRef: jest.fn() }),
+      );
+      expect(shallow(<Editor blockType="AblOck" {...props} />)).toMatchSnapshot();
+    });
+    test.each(Object.values(blockTypes))('renders %p editor when ref is ready', (blockType) => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: { current: 'ref' }, refReady: true, setEditorRef: jest.fn().mockName('setEditorRef') }),
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:36]:
```suggestion
        () => ({
          editorRef: { current: 'ref' },
          refReady: true,
          setEditorRef: jest.fn().mockName('setEditorRef'),
        }),
```

File: src/editors/Editor.test.jsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      );
+      expect(shallow(<Editor blockType={blockType} {...props} />)).toMatchSnapshot();
+    });
+    test('presents error message if no relevant editor found and ref ready', () => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: { current: 'ref' }, refReady: true, setEditorRef: jest.fn().mockName('setEditorRef') }),
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:42]:
```suggestion
        () => ({
          editorRef: { current: 'ref' },
          refReady: true, 
          setEditorRef: jest.fn().mockName('setEditorRef'),
        }),
```

File: src/editors/Editor.test.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+describe('Editor', () => {
+  describe('snapshots', () => {
+    test('renders no editor when ref isnt ready', () => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: null, refReady: false, setEditorRef: jest.fn() }),
+      );
+      expect(shallow(<Editor blockType="AblOck" {...props} />)).toMatchSnapshot();
+    });
+    test.each(Object.values(blockTypes))('renders %p editor when ref is ready', (blockType) => {
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:34]: I hate to say this, but I think this should be split out into separate tests, explicitly spelling out WHICH component is expected to be rendered for each blockType.
What might make sense here, is to do a single snapshot test for a supported type, and then for each type, have a simple test that renders the component and checks the type of the editor component.  The snapshots changing only one field out the larger component are repetitive and harder to verify.

File: src/editors/Editor.test.jsx [hunk 9]
```diff
@@ -35,7 +35,9 @@ describe('Editor', () => {
       hooks.prepareEditorRef.mockImplementationOnce(
         () => ({ editorRef: { current: 'ref' }, refReady: true, setEditorRef: jest.fn().mockName('setEditorRef') }),
       );
-      expect(shallow(<Editor blockType={blockType} {...props} />)).toMatchSnapshot();
+      const wrapper = shallow(<Editor blockType={blockType} {...props} />)
+      expect(wrapper).toMatchSnapshot();
+      expect(wrapper.exists(supportedEditors[blockType])).toBe(true);
```
Comment (connorhaugh) [inline src/editors/Editor.test.jsx:40]: @muselesscreator here is the line I added

File: src/editors/Editor.test.jsx (hunk 8 above)
Comment (muselesscreator) [reply src/editors/Editor.test.jsx:34]: So here, remove the per-blockType snapshot, and add a single test for 1 block type that takes a snapshot, and verifies that the editor spawns once and only once, in the right place.

File: src/editors/Editor.test.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    });
+    test.each(Object.values(blockTypes))('renders %p editor when ref is ready', (blockType) => {
+      hooks.prepareEditorRef.mockImplementationOnce(
+        () => ({ editorRef: { current: 'ref' }, refReady: true, setEditorRef: jest.fn().mockName('setEditorRef') }),
+      );
+      const wrapper = shallow(<Editor blockType={blockType} {...props} />);
+      if(blockType == 'html'){ // snap just one editor to make viewing easier
+        expect(wrapper).toMatchSnapshot();
+      };
+      expect(wrapper.children().children().at(1).is(supportedEditors[blockType])).toBe(true);
```
Comment (muselesscreator) [inline src/editors/Editor.test.jsx:42]: :-)

**Review by muselesscreator — APPROVED** [review]
LGTM

**Pull request merged**